internal/newusers/manager_test.go

```diff
@@ -0,0 +1,577 @@
+
+import (
+	"errors"
+	"flag"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"github.com/ubuntu/authd/internal/newusers"
+	"github.com/ubuntu/authd/internal/newusers/cache"
+	cachetests "github.com/ubuntu/authd/internal/newusers/cache/tests"
+	newusertests "github.com/ubuntu/authd/internal/newusers/tests"
+	"github.com/ubuntu/authd/internal/testutils"
+	"github.com/ubuntu/authd/internal/users"
+)
+
+func TestNewManager(t *testing.T) {
+	t.Parallel()
+
+	tests := map[string]struct {
+		dbFile          string
+		corruptedDbFile bool
+		dirtyFlag       bool
+		markDirty       bool
+
+		expirationDate  string
+		skipCleanOnNew  bool
+		cleanupInterval int
+		procDir         string
+
+		wantErr bool
+	}{
+		"Successfully create a new manager": {},
+
+		// Clean up tests
+		"Clean up all users":  {dbFile: "only_old_users", expirationDate: "2020-01-01"},
+		"Clean up some users": {dbFile: "multiple_users_and_groups", expirationDate: "2020-01-01"},
+		"Clean up also cleans last selected broker for user":     {dbFile: "multiple_users_and_groups_with_brokers", expirationDate: "2020-01-01"},
+		"Clean up on interval":                                   {dbFile: "multiple_users_and_groups", expirationDate: "2020-01-01", cleanupInterval: 1, skipCleanOnNew: true},
+		"Clean up as much as possible if db has invalid entries": {dbFile: "invalid_entries_but_user_and_group1", expirationDate: "2020-01-01"},
+		"Clean up user even if it is not listed on the group":    {dbFile: "user_not_in_groupToUsers", expirationDate: "2020-01-01"},
+		"Do not clean any user":                                  {dbFile: "multiple_users_and_groups"},
+		"Do not clean active user":                               {dbFile: "active_user", expirationDate: "2020-01-01"},
+		"Do not prevent cache creation if cleanup fails":         {dbFile: "multiple_users_and_groups", procDir: "does-not-exist"},
+		"Do not stop cache if cleanup routine fails":             {dbFile: "multiple_users_and_groups", procDir: "does-not-exist", skipCleanOnNew: true, cleanupInterval: 1},
+		"Do not clean user if can not get groups":                {dbFile: "invalid_entry_in_userToGroups", expirationDate: "2020-01-01"},
+		"Do not clean user if can not delete user from group":    {dbFile: "invalid_entry_in_groupByID", expirationDate: "2020-01-01"},
+
+		// Corrupted databases
+		"New recreates any missing buckets and delete unknowns": {dbFile: "database_with_unknown_bucket"},
+		"Database flagged as dirty is cleared up":               {dbFile: "multiple_users_and_groups", dirtyFlag: true},
+		"Corrupted database when opening is cleared up":         {corruptedDbFile: true},
+		"Dynamically mark database as corrupted is cleared up":  {markDirty: true},
+
+		"Error if cacheDir does not exist": {dbFile: "-", wantErr: true},
+	}
+	for name, tc := range tests {
+		tc := tc
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			cacheDir := t.TempDir()
+			if tc.dbFile == "-" {
```

Comment:
using *string instead and keeping this `nil`? Via e.g. `ptrValue`.

Reply:
It would be an anti-pattern with the rest of our tests and it doesn't achieve much since we would still have two cases one to use as the "default" and one to check for empty.